zerver/views/messages.py

```diff
@@ -1469,6 +1470,16 @@ def delete_message_backend(request: HttpRequest, user_profile: UserProfile,
     do_delete_message(user_profile, message)
     return json_success()
 
+@has_request_variables
+def delete_in_topic(request: HttpRequest, user_profile: UserProfile,
+                            stream_id: int=REQ('stream_id'),
+                            topic_name: str=REQ('topic_name')) -> None:
+    # TODO: add req param validators
```

Comment:
`access_stream_by_id` is the main thing you need here.

Reply:
For security -- we need to confirm that the current user has access to the stream in question.